### PR TITLE
Feature/manage species

### PR DESCRIPTION
**Pull request by tejassonar**
Changes in this pull request:
- Added Manage Species Screen
- Stored Scientific species in Realm db
- Local Search functionality

### CODE REVIEW

**Comment by norbertschuler** [comment]
Please check https://sonarcloud.io/project/issues?id=Plant-for-the-Planet-org_treemapper&open=AXc-APsk2kptFZcI8nXW&pullRequest=163&resolved=false&types=BUG

File: package-lock.json [hunk 1]
```diff
@@ -5,85 +5,83 @@
   "requires": true,
   "dependencies": {
     "@babel/code-frame": {
-      "version": "7.8.3",
```
Comment (norbertschuler) [inline package-lock.json:8]: Probably you deleted `package-lock.json` file, so it got created newly and so all available upgrades to all used npm modules got applied. This also implies upgrade of the related Pods for iOS and libraries for Android. 
Either you revert this change and only add `react-native-zip-archive` without upgrading more modules than necessary or you update the iOS Pods with `cd ios; pod install`.

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: app/components/ManageSpecies/index.js [hunk 2]
```diff
@@ -118,30 +129,48 @@ const ManageSpecies = () => {
...
+          }
+        }
+        if (alreadyPresentSpecies.length > 0) {
+          Alert.alert(
+            "Error",
+            `You have added ${alreadyPresentSpecies} already`,
+            [
+              { text: "OK", onPress: () => console.log("OK Pressed") }
+            ],
+            { cancelable: false }
```
Comment (norbertschuler) [inline app/components/ManageSpecies/index.js:172]: Please localize this as e.g. https://github.com/Plant-for-the-Planet-org/treemapper/blob/1678849a4ae3e9b2736ed33f6ae5193f86c1a0a5/app/components/AddSpecies/index.js#L100-L107

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: app/components/ManageSpecies/index.js [hunk 3]
```diff
@@ -163,11 +144,12 @@ const ManageSpecies = () => {
           }
         }
         if (alreadyPresentSpecies.length > 0) {
+          console.log(alreadyPresentSpecies,'already Present................');
           Alert.alert(
             "Error",
             `You have added ${alreadyPresentSpecies} already`,
             [
-              { text: "OK", onPress: () => console.log("OK Pressed") }
+              { text: "OK", onPress: () => {console.log("OK Pressed"); setAlreadyPresentSpecies([]);} }
```
Comment (sagararyal) [inline app/components/ManageSpecies/index.js]: I believe with the New UI, you'll never come across this issue.

File: app/components/SelectSpecies/index.js [hunk 4]
```diff
@@ -333,16 +340,21 @@ const SelectSpecies = ({
 
   const onPressTreeCountNextBtn = () => {
     let speciesListClone = speciesList;
-    let specie = speciesListClone[activeSpeice];
+    console.log(typeof speciesList, 'speciesList Type')
+    let specie = speciesListClone[index];
     specie.treeCount = Number(treeCount) ? treeCount : undefined;
-    speciesListClone.splice(activeSpeice, 1, specie);
+    console.log(speciesListClone,specie.treeCount,'specie.treeCount', index, specie.scientificName);
+    speciesListClone.splice(index, 1, specie);
```
Comment (norbertschuler) [inline app/components/SelectSpecies/index.js:347]: App crashes for me when trying to selected a species for my registered trees:
Simulator Screen Shot - iPhone 11 - 2021-01-28 at 10 46 20

Comment (norbertschuler) [reply app/components/SelectSpecies/index.js:347]: This happens after entering the number of planted trees having chosen to register "Multiple Trees" + "Off Site - Point":
Simulator Screen Shot - iPhone 11 - 2021-01-28 at 10 52 37

**Comment by norbertschuler** [comment]
- [ ] I already translated the version of this page used in the develop branch - please copy that translations and add further ones to also localize this new page:

New:
Simulator Screen Shot - iPhone 11 - 2021-01-28 at 10 58 05

Old:
Simulator Screen Shot - iPhone 11 - 2021-01-28 at 11 02 55Simulator Screen Shot - iPhone 11 - 2021-01-28 at 11 05 22

**Comment by norbertschuler** [comment]
- [ ] Any change to delete falsely selected species again?

**Comment by tejassonar** [comment]
- [ ] reomve keyboard after clicking on 'x' icon.
image

File: app/components/SelectSpecies/index.js (hunk 4 above)
Comment (ankitecd) [reply app/components/SelectSpecies/index.js]: Creating a different branch to solve multiple trees issue.

**Comment by norbertschuler** [comment]
- [ ] I do not yet completely understand the UI workflow of the new implementation. e.g. the following two things:

Simulator Screen Shot - iPhone 11 - 2021-01-29 at 11 44 01

**Comment by norbertschuler** [comment]
- [ ] when I choose three species, but only add a tree number for the first and then click the green button to save and go on, the app crashes or me (iPhone 11 simulator):

Simulator Screen Shot - iPhone 11 - 2021-01-29 at 11 46 32Simulator Screen Shot - iPhone 11 - 2021-01-29 at 11 46 26

**Review by norbertschuler — APPROVED** [review]
- Single tree registration is working
- Multiple tree registration has issues (including crash)
- Managing species in profile menu also has issue (non functioning buttons)
